CONTRIBUTING.md

```diff
@@ -1,54 +1,151 @@
+4. Adding new or improved functionality to the existing codebase.
 
-2. Create conda environment. For example:
+Items 2-4 require setting up a local development environment, see [Local development steps](#Contributing-code-via-pull-requests) for more information.
 
-```shell
-conda create -n pymc_marketing_env
-```
+## Opening issues
+We appreciate being notified of problems with the existing PyMC-Marketing code. We prefer that issues be filed the on [Github Issue Tracker](https://github.com/pymc-labs/pymc-marketing/issues), rather than on social media or by direct email to the developers.
 
-3. Activate environment.
+Please verify that your issue is not being currently addressed by other issues or pull requests by using the GitHub search tool to look for key words in the project issue tracker.
 
-```shell
-conda activate pymc_marketing_env
-```
+## Contributing code via pull requests
 
-4. Install `pymc_marketing` package:
+While issue reporting is valuable, we strongly encourage users who are inclined to do so to submit patches for new or existing issues via pull requests. This is particularly the case for simple fixes, such as typos or tweaks to documentation, which do not require a heavy investment of time and attention.
 
-```shell
-make init
-```
+Contributors are also encouraged to contribute new code to enhance PyMC-Marketing's functionality, via pull requests.
 
-5. To run tests:
+The preferred workflow for contributing to PyMC-Marketing is to fork the GitHub repository, clone it to your local machine, and develop on a feature branch.
 
-```shell
-make test
-```
+For more instructions see the [Pull request checklist](#pull-request-checklist)
 
-6. To check code style:
+## Local development steps
+
+1. If you have not already done so, fork the [project repository](https://github.com/pymc-labs/pymc-marketing) by clicking on the 'Fork' button near the top right of the main repository page. This creates a copy of the code under your GitHub user account.
+
+1. Clone your fork of the `pymc-marketing` repo from your GitHub account to your local disk, and add the base repository as a remote:
+
+    ```bash
+    git clone git@github.com:<your GitHub handle>/pymc-marketing.git
+    cd pymc-marketing
+    git remote add upstream git@github.com:pymc-labs/pymc-marketing.git
+    ```
+
+1. Create a feature branch (e.g. `my-feature`) to hold your development changes:
+
+   ```bash
+   git checkout -b my-feature
+   ```
+
+   Always use a feature branch. It's good practice to never routinely work on the `main` branch of any repository.
+
+1. Create an environment. For example:
+
+    ```bash
+    conda create -n pymc_marketing_env
+    ```
+
+    Activate the environment.
+
+    ```bash
+    conda activate pymc_marketing_env
+    ```
+
+    Install the package (in editable mode) and its development dependencies:
+
+    ```bash
+    make init
+    ```
+
+    Set [pre-commit hooks](https://pre-commit.com/):
```

Comment:
they need to install `pre-commit` inside the env first. Shall we suggest `make check_lint` first? or just install `pre-commit`?

Reply:
Good point. Have added something about this